pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -143,7 +143,7 @@ func (a *Actuator) updateMachineStatus(machine *clusterv1.Machine, awsStatus *pr
 		time := metav1.Now()
 		machineCopy.Status.LastUpdated = &time
 
-		if err := a.client.Update(context.Background(), machineCopy); err != nil {
+		if err := a.client.Status().Update(context.Background(), machineCopy); err != nil {
```

Comment:
We can not call `Status()` until we start running/testing the installer on top of OpenShift that vendors kubernetes 1.12. Given a machine definition is CRD based, updating status needs additional permissions which are disabled in 1.11 by default.

Reply:
Which version of Kubernetes did you run?

Reply:
Ok, minikube running inside the CI is actually deploying kubernetes 1.10. I fixed that and bump k8s to 1.11.3. Still, I  am getting the following error message:
```
error updating machine status: the server could not find the requested resource (put machines.cluster.k8s.io pr-247-87-worker-machineset-9869b6-26fmd)
```

Which based on [1] is related to updating CRD status subresource. I will keep investigating.

[1] https://github.com/kubernetes/code-generator/issues/49

Reply:
I am even able to reproduce this locally with minikube while deploying kubernetes 1.11.3. Still getting:
```
the server could not find the requested resource (put machines.cluster.k8s.io...
```

1. `minikube start --vm-driver=kvm2 --kubernetes-version v1.11.3 --v 5`
2. deploying manifests from https://github.com/openshift/machine-api-operator/tree/master/owned-manifests
    manager image set to docker.io/openshift/origin-aws-machine-controllers:v4.0.0
3. deploying cluster and machine
4. running the machine controller locally pointing to `/root/.kube/config`.

@vikaschoudhary16 Can you verify if you can reproduce it locally as well? Just with pure kubernetes?

Reply:
@ingvagabund found that error you seeing and error in the e2e test were same and that was because of incorrect machine crd manifest being used in e2e. It was missing status subresource.